### PR TITLE
plan: fix name with white space at the end

### LINKED ISSUE #3835: No table name validity check in drop table if exists
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

```sql
drop table if exists ``;
```

2. What did you expect to see?

ERROR 1103 (42000): Incorrect table name ''

3. What did you see instead?

Query OK, 0 rows affected (0.00 sec)

4. What version of TiDB are you using (`tidb-server -V`)?

### PR DESCRIPTION
**Pull request by jackysp**
Fix #3835 #3836 #3838 
PTAL @winkyao @zimulala @hicqu

### CODE REVIEW

File: plan/validator.go [hunk 1]
```diff
@@ -353,3 +436,14 @@ func checkFieldLengthLimitation(colDef *ast.ColumnDef) error {
 	}
 	return nil
 }
+
+// See https://dev.mysql.com/doc/refman/5.7/en/identifiers.html
+func isIncorrectName(name string) bool {
```
Comment (shenli) [inline plan/validator.go]: Please add a unit test for this function.

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
@winkyao PTAL

**Comment by jackysp** [comment]
Race fixed. @shenli PTAL

File: plan/validator_test.go [hunk 2]
```diff
@@ -75,7 +75,7 @@ func (s *testValidatorSuite) TestValidator(c *C) {
 		{"alter table t add column c int auto_increment key, auto_increment=10", true,
 			errors.New("[autoid:3]No support for setting auto_increment using alter_table")},
 		{"alter table t add column c int auto_increment key", true, nil},
-		{"create table `` (a int)", true, errors.New("[ddl:1103]Incorrect table name ''")},
+
```
Comment (zimulala) [inline plan/validator_test.go]: Remove this blank line.

File: ddl/ddl.go [hunk 3]
```diff
@@ -111,8 +110,13 @@ var (
 	ErrInvalidOnUpdate = terror.ClassDDL.New(codeInvalidOnUpdate, "invalid ON UPDATE clause for the column")
 	// ErrTooLongIdent returns for too long name of database/table/column.
 	ErrTooLongIdent = terror.ClassDDL.New(codeTooLongIdent, "Identifier name too long")
-	// ErrWrongTableName return for wrong table name.
-	ErrWrongTableName = terror.ClassDDL.New(codeWrongTableName, "Incorrect table name '%s'")
+
```
Comment (zimulala) [inline ddl/ddl.go]: Remove this blank line.

**Comment by zimulala** [comment]
Rest LGTM

File: plan/validator.go (hunk 1 above)
Comment (jackysp) [inline plan/validator.go]: All statements have been covered by test. I think we can add a single unit test if this function become more complex in future.

File: ddl/ddl.go (hunk 3 above)
Comment (jackysp) [inline ddl/ddl.go]: Done.

File: plan/validator_test.go (hunk 2 above)
Comment (jackysp) [inline plan/validator_test.go]: Done.

**Comment by zimulala** [comment]
LGTM

**Pull request merged**